<launch>
    <arg name="xacro_path"   default="$(find ur5_allegro_moveit)/models/ur5_allegro.xacro" />
    <arg name="urdf_args"   default=""/>
    <arg name="load_xacro"   default="true"/>
    <arg name="viz"   default="true"/>
    <arg name="start_leap"   default="true"/>
    <arg name="leap_topic"   default="/leap_motion/leap_filtered"/>
    <arg name="left_handed"   default="False"/>
    <arg name="joint_state_topic"   default="joint_states"/>
    <arg name="joint_state_gui"   default="false"/>


    <group if="$(arg start_leap)">
        <include file="$(find relaxed_leap_teleop)/launch/leap_start.launch">
            <arg name="leap_camera_mode"   value="False"/>
        </include>
    </group>

    

    <group if="$(arg load_xacro)">
        <param name="robot_description" command="xacro --inorder '$(arg xacro_path)' $(arg urdf_args)"/>
        <node pkg="tf2_ros" type="static_transform_publisher" name="base2common_world" args="0 0 0 0 0 0 1 base common_world" />
    </group>

    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" ns="/" >
        <param name="publish_frequency" type="double" value="60.0" />
        <param name="tf_prefix" type="string" value="" />
    </node>

    <!-- <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
            <param name="use_gui" value="$(arg joint_state_gui)"/>
            <rosparam param="source_list">[merged_states]</rosparam>
            <rosparam param="zeros">
            shoulder_pan_joint: -1.085
            shoulder_lift_joint: -1.158
            elbow_joint: 1.924
            wrist_1_joint: 3.868
            </rosparam>
    </node> -->

    <!-- <node pkg="relaxed_leap_teleop" type="leap_to_ur5.py" name="ur5_teleop" output="screen"> -->
        <!-- <param name="leap_topic" type="str" value="$(arg leap_topic)" /> -->
        <!-- <param name="max_msg_age_in_sec" type="double" value="0.3" /> -->
        <!-- <param name="consumption_rate" type="double" value="30" /> -->
        <!-- <param name="lefthand_mode" type="bool" value="$(arg left_handed)" /> -->
        <!-- <param name="debug_mode" type="bool" value="true" /> -->
        <!-- <remap from="joint_states" to="$(arg joint_state_topic)"/> -->
    <!-- </node> -->

    <group if="$(arg viz)">
        <node name="teleop_rviz" pkg="rviz" type="rviz" args="-d $(find relaxed_leap_teleop)/rviz/teleop_viz.rviz" />
    </group>

</launch>